### PR TITLE
Pollster final review

### PR DESCRIPTION
**Pull request by agyen**

### CODE REVIEW

File: mailgun.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+}
+ 
+const nodemailerMailgun = nodemailer.createTransport(mg(auth));
+
+async function start(toEmail) {
+  
+nodemailerMailgun.sendMail({
+  from: toEmail.creator_email,
+  to: toEmail.recipient_email, 
+//   cc:'second@domain.com',
```
Comment (FrancisBilla) [inline mailgun.js]: remove comments, trust your code

File: mailgun.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                      <td style="font-family: sans-serif; font-size: 14px; vertical-align: top; background-color: #3498db; border-radius: 5px; text-align: center;"> <a href = "https://polls-angular-ui.herokuapp.com/thankyoupage/?option_id=`+options.option_id+`&poll_id=`+options.poll_id+`" target="_blank" style="display: inline-block; color: #ffffff; background-color: #3498db; border: solid 1px #3498db; border-radius: 5px; box-sizing: border-box; cursor: pointer; text-decoration: none; font-size: 14px; font-weight: bold; margin: 0; padding: 12px 25px; text-transform: capitalize; border-color: #3498db;"> `+options.content+` </a> </td>
+                    </tr>`
+                    ))+`
+                    </tbody>
+                  </table>
+                  </td>
+                </tr>
+                </tbody>
+              </table>
+
```
Comment (FrancisBilla) [inline mailgun.js]: remove white spaces

File: src/app/thankyoupage/thankyoupage.component.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  ngOnInit() {
+    setTimeout(() => {
+      this.activatedRoute.queryParams.
+        subscribe(params => {
+          this.responses.option_id = params.option_id
+          this.responses.poll_id = params.poll_id
+
+          this.responseService.recordResponse(this.responses).subscribe(response => {
+            console.log(response);
```
Comment (FrancisBilla) [inline src/app/thankyoupage/thankyoupage.component.ts]: avoid console logging

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+dist: trusty
+sudo: false
+
+language: node_js
+node_js:
+  - '10'
+
+before_install:
```
Comment (sammoorhouse) [inline .travis.yml]: remove test framework if you're not using it

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+cache:
+  directories:
+    - ./node_modules
+
+install:
+  - npm install
+
+script:
+  - npm run test -- --watch=false --no-progress --browsers=ChromeHeadlessNoSandbox
```
Comment (sammoorhouse) [inline .travis.yml]: you don't have any tests :)

File: karma.conf.js [hunk 6]
```diff
@@ -21,6 +21,12 @@ module.exports = function (config) {
       fixWebpackSourcePaths: true
     },
     reporters: ['progress', 'kjhtml'],
+    customLaunchers: {
+      ChromeHeadlessNoSandbox: {
```
Comment (sammoorhouse) [inline karma.conf.js]: karma is a JS unit testing framework, if you're not using it you ought to kill this code :)

File: mailgun.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+ 
+const nodemailerMailgun = nodemailer.createTransport(mg(auth));
+
+async function start(toEmail) {
+  
+nodemailerMailgun.sendMail({
+  from: toEmail.creator_email,
+  to: toEmail.recipient_email, 
+  subject: 'Turntabl Poll',
+  'h:Reply-To': 'isaac.agyen@turntabl.io',
```
Comment (sammoorhouse) [inline mailgun.js]: investigate 'h:' - why isn't this just reply-to

File: mailgun.js [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+        </div>
+      </td>
+      <td style="font-family: sans-serif; font-size: 14px; vertical-align: top;">&nbsp;</td>
+      </tr>
+    </table>
+    </body>
+  </html>`,
+}, (err, info) => {
+  if (err) {
+    console.log(`Error: ${err}`);
```
Comment (sammoorhouse) [inline mailgun.js]: who are these designed for? start to use the javascript debugger to investigate success and failure of client-side code.

File: package.json [hunk 9]
```diff
@@ -3,34 +3,53 @@
   "version": "0.0.0",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
+    "start": "node server.js",
     "build": "ng build",
```
Comment (sammoorhouse) [inline package.json]: remove unused scripts

File: package.json [hunk 10]
```diff
@@ -3,34 +3,53 @@
...
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.803.20",
-    "@angular/cli": "~8.3.20",
+    "@angular-devkit/build-angular": "^0.803.24",
+    "@angular-devkit/core": "^8.3.23",
+    "@angular/cli": "^8.3.23",
     "@angular/compiler-cli": "~8.2.14",
     "@angular/language-service": "~8.2.14",
-    "@types/node": "~8.9.4",
     "@types/jasmine": "~3.3.8",
```
Comment (sammoorhouse) [inline package.json]: you're probably not using these deps, consider finding out how to identify & remove unused dependencies

File: server.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
+const express = require('express');
+const path = require('path');
+const cors = require('cors');
+const bodyParser = require('body-parser');
+const nodemailer = require('nodemailer');
+const mail = require('./mailgun');
+const poll = process.env.POLL_BACKEND_URL;
+const options = process.env.OPTION_BACKEND_URL;
+const sendmail_url = process.env.SENDMAIL_URL
+const RESPONSE_URL = process.env.RESPONSE_URL
```
Comment (sammoorhouse) [inline server.js]: consider casing and naming here

File: src/app/app-routing.module.ts [hunk 12]
```diff
@@ -1,11 +1,20 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import {MatpollformComponent } from './matpollform/matpollform.component';
```
Comment (sammoorhouse) [inline src/app/app-routing.module.ts]: consider spacing - make sure your IDE is setup to prettify code on save.

File: src/app/header/header.component.html [hunk 13]
```diff
@@ -0,0 +1,27 @@
+<mat-toolbar color="primary">
+  <button mat-button (click)="addPoll()" routerLink="/matpollform" title="Create new poll">
+    <mat-icon>add</mat-icon>
+  </button>
+  <span class="application-name" (click)="goToRoot()">Pollster</span>
+
+  <mat-menu #appMenu="matMenu">
```
Comment (sammoorhouse) [inline src/app/header/header.component.html]: let's remove this until we need it

File: src/app/thankyoupage/thankyoupage.component.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+  selector: 'app-thankyoupage',
+  templateUrl: './thankyoupage.component.html',
+  styleUrls: ['./thankyoupage.component.scss']
+})
+export class ThankyoupageComponent implements OnInit {
+
+  constructor(private responseService: ResponseService, private activatedRoute: ActivatedRoute,
+    private location: Location) { }
+    
+    responses = {
```
Comment (sammoorhouse) [inline src/app/thankyoupage/thankyoupage.component.ts]: this ought to be a typescript type

File: src/assets/script/app.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+let newOption = document.querySelector("#newOptionbtn")
+
+console.log("js working...");
+
+newOption.addEventListener('click', addNewOption);
+
+const addNewOption = () => {
+    console.log("Adding new option...");
```
Comment (sammoorhouse) [inline src/assets/script/app.js]: let's remove these stray client-side console logs

File: src/assets/script/app.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
+let newOption = document.querySelector("#newOptionbtn")
+
+console.log("js working...");
+
+newOption.addEventListener('click', addNewOption);
```
Comment (sammoorhouse) [inline src/assets/script/app.js]: let's remove this event handler that just logs :)

File: src/index.html [hunk 17]
```diff
@@ -3,11 +3,68 @@
 <head>
   <meta charset="utf-8">
   <title>PollUi</title>
+
+  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-118037368-1"></script>
```
Comment (sammoorhouse) [inline src/index.html]: do we need this?

File: src/main.ts [hunk 18]
```diff
@@ -1,3 +1,4 @@
+import 'hammerjs';
```
Comment (sammoorhouse) [inline src/main.ts]: hammer is for touchscreens, do we need that?

File: mailgun.js (hunk 7 above)
Comment (YaaNuamah) [reply mailgun.js]: With the 'h:Reply-To each user gets a unique hash in the reply-to field so I know who is replying. Basically the reply-to looks like this:

"h:Reply-To" : "inboundaddress+hash@mydomain.com"

Based on official Mailgun API docs here [sending](https://documentation.mailgun.com/api-sending.html#) you can specify the Reply-to option, but it must be defined as h:X-My-Header:

File: .travis.yml (hunk 4 above)
Comment (agyen) [reply .travis.yml]: its needed for travis to build, travis failed when we took it out

File: .travis.yml (hunk 5 above)
Comment (YaaNuamah) [reply .travis.yml]: Yes it is needed to build successfully on Travis , without it the app fails on Travis. We tried removing it and searched on it. [check this out](https://medium.com/faun/configuring-travis-ci-for-angular-application-34afee1715f)

File: src/main.ts [hunk 19]
```diff
@@ -9,4 +9,4 @@ if (environment.production) {
 }
 
 platformBrowserDynamic().bootstrapModule(AppModule)
-  .catch(err => console.error(err));
+  .catch();
```
Comment (sammoorhouse) [inline src/main.ts]: can probably remove .catch()

File: karma.conf.js (hunk 6 above)
Comment (YaaNuamah) [reply karma.conf.js]: okay removed and running successfully.

File: package.json (hunk 10 above)
Comment (YaaNuamah) [reply package.json]: okay removed unused dependencies.

File: src/main.ts (hunk 19 above)
Comment (YaaNuamah) [reply src/main.ts]: okay removed.

File: .prettierrc [hunk 20]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (sammoorhouse) [inline .prettierrc]: lets remove this please

**Pull request merged**